x-pack/filebeat/fbreceiver/receiver_test.go

```diff
@@ -206,3 +206,57 @@ func TestMultipleReceivers(t *testing.T) {
 		},
 	})
 }
+
+func TestUniqueLoggerNamePerReceiver(t *testing.T) {
+	config := Config{
+		Beatconfig: map[string]interface{}{
+			"filebeat": map[string]interface{}{
+				"inputs": []map[string]interface{}{
+					{
+						"type":    "benchmark",
+						"enabled": true,
+						"message": "test",
+						"count":   1,
+					},
+				},
+			},
+			"output": map[string]interface{}{
+				"otelconsumer": map[string]interface{}{},
+			},
+			"logging": map[string]interface{}{
+				"level": "debug",
+				"selectors": []string{
+					"*",
+				},
+			},
+			"path.home": t.TempDir(),
+		},
+	}
+
+	oteltest.CheckReceivers(oteltest.CheckReceiversParams{
+		T: t,
+		Receivers: []oteltest.ReceiverConfig{
+			{
+				Name:    "r1",
+				Config:  &config,
+				Factory: NewFactory(),
+			},
+			{
+				Name:    "r2",
+				Config:  &config,
+				Factory: NewFactory(),
+			},
+		},
+		AssertFunc: func(t *assert.CollectT, logs map[string][]mapstr.M, zapLogs *observer.ObservedLogs) {
+			require.Conditionf(t, func() bool {
+				return len(logs["r1"]) > 0 && len(logs["r2"]) > 0
+			}, "expected receivers to have logs, got logs: %v", logs)
+
+			r1StartLogs := zapLogs.FilterMessageSnippet("Starting input").FilterField(zap.String("name", "r1"))
+			assert.Equal(t, 1, r1StartLogs.Len(), "r1 should have a single start log")
+
+			r2StartLogs := zapLogs.FilterMessageSnippet("Starting input").FilterField(zap.String("name", "r2"))
+			require.Equal(t, 1, r2StartLogs.Len(), "r2 should have a single start log")
```

Comment:
I see the `CheckReceivers` [creating a logger](https://github.com/elastic/beats/blob/8d31036ab348100d9f1cdab6dc7e086e89b0f509/libbeat/otelbeat/oteltest/oteltest.go#L86), so I'm trying to understand how it tests your changes. 
If I run this test on `main`, would it fail because the logger used isn't local, is it the idea?

Reply:
Here is the rationale behind this test and an explanation how it is testing this change https://github.com/elastic/ingest-dev/issues/5202#issuecomment-2724971562.

Reply:
Basically, the name field in the global logger was the name of the last receiver created. The "Starting input" log message should be unique, so one for each receiver. If the bug is still present then the logger with name r2 should have two "Starting input" logs. Perhaps we can write a short description on this test to explain the rationale.

Reply:
I see. Perhaps a comment on the test with a tl;dr of how or why this test ensures the local logger is used would be good.

Reply:
+1 to adding a comment explaining how the test should work